### ISSUE TITLE
[Bug]: UnicodeDecodeError

### ISSUE THREAD

**Issue by perseus177** [custom-components/ble_monitor#1257]
### What happened?

**Error in log file** 

## System Information

HA version | core-2023.10.5
-- | --
Passive BLE monitor integration version | 12.6.1
installation_type | Home Assistant Supervised
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.5
os_name | Linux
os_version | 5.10.103-v7l+
arch | armv7l
timezone | Europe/Prague
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.33.0
Stage | running
Available Repositories | 1324
Downloaded Repositories | 12

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Raspbian GNU/Linux 10 (buster)
-- | --
update_channel | stable
supervisor_version | supervisor-2023.10.1
agent_version | null
docker_version | 24.0.6
disk_total | 109.8 GB
disk_used | 11.2 GB
healthy | true
supported | failed to load: Unsupported
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.6.0), MariaDB (2.6.1), Mosquitto broker (6.3.1), TasmoAdmin (0.25.1), Nginx Proxy Manager (0.12.3), Zigbee2MQTT (1.33.1-1), Samba share with features (10.0.2.3)

</details>

<details><summary>Dashboards</summary>

dashboards | 1
-- | --
resources | 5
views | 11
mode | yaml

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 26. októbra 2023 o 16:55
-- | --
current_recorder_run | 26. októbra 2023 o 18:55
estimated_db_size | 444.02 MiB
database_engine | sqlite
database_version | 3.41.2

</details>

### Sensor type

- LYWSD03MMC
- LYWSDCGQ
- RTCGQ02LM
- YLYK01YL
- YM-K1501

### Relevant log output

```shell
Logger: homeassistant
Source: custom_components/ble_monitor/ble_parser/__init__.py:139
Integration: Passive BLE monitor (documentation, issues)
First occurred: 28. októbra 2023 o 15:43:05 (2 occurrences)
Last logged: 28. októbra 2023 o 16:03:54

Error doing job: Fatal error: protocol.data_received() call failed.
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 1003, in _read_ready__data_received
    self._protocol.data_received(data)
  File "/usr/local/lib/python3.11/site-packages/aioblescan/aioblescan.py", line 1855, in data_received
    self.process(packet)
  File "/config/custom_components/ble_monitor/__init__.py", line 586, in process_hci_events
    sensor_msg, tracker_msg = self.ble_parser.parse_raw_data(data)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/ble_monitor/ble_parser/__init__.py", line 139, in parse_raw_data
    shortened_local_name = adstruct[2:].decode("utf-8")
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc3 in position 12: unexpected end of data
```

**Comment by Ernst79** [comment]
This is most likely caused by a malfunctioning Bluetooth dongle or interference. 

In this issue it was due to interference of two sensors: https://github.com/custom-components/ble_monitor/issues/1177

Or it can be a firmware issue, check this issue https://github.com/custom-components/ble_monitor/issues/870 with this reference to a fix. 

> I found the problem, because this problem was reported in dmesg:
> Bluetooth: hci0: Frame reassembly failed (-84).
> And I found the fix https://github.com/raspberrypi/firmware/issues/1150.

**Comment by perseus177** [comment]
Additional info about Hardware 
RPI4 - 8GB RAM with official power supply
- Boot from SSD on USB extension cable (1.5M from RPI)
- BLE integrated on RPI (not separate dongle)

**Comment by flouck** [comment]
I have a similar log entry:

Cette erreur provient d'une intégration personnalisée
```
Logger: homeassistant
Source: custom_components/ble_monitor/ble_parser/__init__.py:142
Integration: Passive BLE monitor (documentation, issues)
First occurred: 27 novembre 2023 à 19:55:11 (1 occurrences)
Last logged: 27 novembre 2023 à 19:55:11

Error doing job: Fatal error: protocol.data_received() call failed.
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 1003, in _read_ready__data_received
    self._protocol.data_received(data)
  File "/usr/local/lib/python3.11/site-packages/aioblescan/aioblescan.py", line 1855, in data_received
    self.process(packet)
  File "/config/custom_components/ble_monitor/__init__.py", line 586, in process_hci_events
    sensor_msg, tracker_msg = self.ble_parser.parse_raw_data(data)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/ble_monitor/ble_parser/__init__.py", line 142, in parse_raw_data
    complete_local_name = adstruct[2:].decode("utf-8")
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf7 in position 7: invalid start byte
```

**Comment by Ernst79** [comment]
This error should be fixed in 12.6.3
